server/service/hosts.go

```diff
@@ -928,6 +928,13 @@ func (svc *Service) OSVersions(ctx context.Context, teamID *uint, platform *stri
 
 	osVersions, err := svc.ds.OSVersions(ctx, teamID, platform)
 	if err != nil && fleet.IsNotFound(err) {
+		// only return not found if invalid team id, otherwise API should return empty JSON array
+		if teamID != nil {
+			_, err := svc.ds.Team(ctx, *teamID)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
This "team exists" check should be performed before the call to `svc.ds.OSVersions`.

Reply:
This is meant to handle the case where the team was added after UpdateOSVersions was ran. Most of the time, the team should exist. Adding a check for the team before calling `svc.ds.OSVersions` would mean that we now perform an extra db call *every* time.

Reply:
Ah, makes sense, let's add the explanation as a comment.